mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -449,6 +453,52 @@ fn init_wasm_event_streaming(ctx: &MmArc) {
     }
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+async fn init_db_migration_watcher_loop(ctx: MmArc) {
+    use std::collections::HashSet;
+
+    let mut migrations = HashSet::new();
+    let receiver = &ctx
+        .init_db_migration_watcher()
+        .expect("db_m igration_watcher initialization failed");
+    let mut guard = receiver.lock().await;
+
+    while let Some(ids) = guard.next().await {
+        if migrations.contains(&ids.db_id) {
+            debug!("{} migrated, skipping migration..", ids.db_id);
+            continue;
+        }
+
+        // run db migration for db_id if new activated pubkey is unique.
+        if let Err(err) = run_db_migration_impl(&ctx, Some(&ids.db_id), Some(&ids.shared_db_id)).await {
+            error!("{err:?}");
+            continue;
+        };
+
+        // insert new db_id to migration list
+        migrations.insert(ids.db_id.clone());
+        // Fetch and extend ctx.coins_needed_for_kick_start from new intialized db.
+        if let Err(err) = kick_start(ctx.clone(), Some(&ids.db_id)).await {
+            error!("{err:?}");
+            continue;
+        };
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+async fn run_db_migration_impl(ctx: &MmArc, db_id: Option<&str>, shared_db_id: Option<&str>) -> MmInitResult<()> {
+    fix_directories(ctx, db_id, shared_db_id)?;
+    AsyncSqliteConnPool::init(ctx, db_id)
+        .await
+        .map_to_mm(MmInitError::ErrorSqliteInitializing)?;
+    SqliteConnPool::init(ctx, db_id).map_to_mm(MmInitError::ErrorSqliteInitializing)?;
+    SqliteConnPool::init_shared(ctx, db_id).map_to_mm(MmInitError::ErrorSqliteInitializing)?;
```

Comment:
By a chance, should `shared_db_id` not be in init_shared() instead of `db_id`?

Reply:
thanks for the catch again!